docs/architecture-decisions/adr002-default-catalog-file-format.md

```diff
@@ -0,0 +1,245 @@
+
+<table>
+  <thead>
+    <tr>
+      <th>Field</th>
+      <th>Type</th>
+      <th>Required</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td valign="top"><code>name</code></td>
+      <td valign="top">String</td>
+      <td valign="top">Yes</td>
+      <td valign="top">
+        <p>
+          The name of the entity. Names must be unique per kind, within a given namespace.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>namespace</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          The `name` of a namespace that the entity belongs to. This field is optional.
+        </p>
+        <p>
+          Namespaces may also be part of the catalog, and are <code>v1</code>/<code>Namespace</code>
+          objects, i.e. not Backstage specific but the same as in Kubernetes.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>description</code></td>
+      <td valign="top">String</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          A human readable description of the entity, to be shown in Backstage. May contain
+          markdown.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>labels</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          Labels are key/value pairs of that are attached to the entity, and their use is
+          identical to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/">kubernetes object labels</a>.
+        </p>
+      </td>
+    </tr>
+    <tr>
+      <td valign="top"><code>annotations</code></td>
+      <td valign="top">Object</td>
+      <td valign="top">No</td>
+      <td valign="top">
+        <p>
+          An object with arbitrary non-identifying metadata attached to the entity,
+          identical in use to <a href="https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/">kubernetes object annotations</a>.
+        </p>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+The precise structure of the `spec` root object depends on the `apiVersion` and
+`kind` combination, and some kinds may not even have a `spec` at all. See
+further down in this document for the specification structure of specific kind.
+
+## Entity Specification: backstage.io/v1 Component
+
+| Field        | Value             |
+| ------------ | ----------------- |
+| `apiVersion` | `backstage.io/v1` |
+| `kind`       | `Component`       |
+
+The `spec` object for this kind is as follows:
+
+| Field  | Type   | Required | Description                            |
+| ------ | ------ | -------- | -------------------------------------- |
+| `type` | String | Yes      | The type of component, e.g. `service`. |
+
+TODO: More fields
+
+## TODO: More entity specifications
```

Comment:
The other two that came out of the system model exercise were `API` and `Resource`. They probably end up being quite similar to the `Component` one when it comes to the metadata part (which is what we would want), except that APIs might come with an inline API specification and resources might have a URL in the spec. Could try to flesh that out a bit, or leave it for later if we want to start with Components only for now?

Reply:
I think it would be great to start at least think in broad strokes about the other ones too. One thing that comes to mind is how they are named/identified - will an API have a `name` at all? Who picks it? How is it scoped?

Will a resource have a `name`, and does it come from the external system? Inherited from Google for example, or a semantically useful name for the reader?

Will API specs normally at all be written in this "yaml form"? Or are they basically only useful to define as REST API objects? Will a service component YAML file typically reference e.g. an OpenAPI spec file next to it and then the ingester is responsible for being smart enough to pick up that one too and ingest it along with the component itself? Etc

Reply:
Those are good questions, and I'm not entirely sure.

If we just consider indexing yaml files from VCS for now, it'd be the same as for `Component`: you'd need to pick a name when you create the file in git. Scoping would be the same also, I'd enforce that the `name` is unique within the kind (and namespace).

The interesting question is how we would extend it to things that are not declared in files and checked in, but either inferred from other files (say, a gRPC API from a protobuf directory in a repository) or periodically synced form an external system (eg cloud resources from AWS, GCP, ...). That glue will need to handle naming etc. 

But at this point, maybe it's fine to only look at describing it manually in files on disk? The same pattern of extending it to be inferred or indexed at runtime applies to the Components also....